express-error-handling/server/server.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import 'dotenv/config';
+import pg from 'pg';
+import express from 'express';
+import ClientError from './client-error.js';
+import errorMiddleware from './error-middleware.js';
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades"
+    `;
+    const result = await db.query(sql);
+    const grades = result.rows;
+    res.json(grades);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course } = req.body;
+    const score = Number(req.body.score);
+    if (!Number.isInteger(score) || score < 0 || score > 100) {
+      throw new ClientError(400, 'score must be an integer between 0 and 100');
+    }
+    if (!name || !course) {
+      throw new ClientError(400, 'name, course, and score are required fields');
+    }
+    const sql = `
+      insert into "grades" ("name", "course", "score")
+      values ($1, $2, $3)
+      returning *
+    `;
+    const params = [name, course, score];
+    const result = await db.query(sql, params);
+    const [newGrade] = result.rows;
+    res.status(201).json(newGrade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId < 1) {
+      throw new ClientError(400, `${gradeId} must be a positive integer`);
+    }
+    const sql = `
+      select *
+        from "grades"
+      where "gradeId" = $1
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const [grade] = result.rows;
+    if (!grade) {
+      throw new ClientError(404, `cannot find grade with gradeId ${gradeId}`);
+    } else {
```

Comment:
This else is redundant since when you throw execution stops. So the only way to get to line 73 is if you do not throw.